### PR TITLE
[Performance] Defer some calculations only when they are needed for TextField

### PR DESCRIPTION
**Pull request by Effanuel**
## Description
Improving performance of TextField, by optimizing ValidationMessage
*Enter description to help the reviewer understand what's the change about...*

## Changelog
* Defer calculation of validation message only when it's needed

### CODE REVIEW

File: src/incubator/TextField/CharCounter.tsx [hunk 1]
```diff
@@ -7,12 +7,15 @@ import {CharCounterProps} from './types';
 
 const CharCounter = ({maxLength, charCounterStyle, testID}: CharCounterProps) => {
   const {value} = useContext(FieldContext);
+
+  const textStyle = useMemo(() => [styles.container, charCounterStyle], [charCounterStyle]);
```
Comment (ethanshar) [inline src/incubator/TextField/CharCounter.tsx:11]: This is a redundant memoization. when value changes, the char counter always changes and re-redner, so u might as well pass a new style. 

No reason to memoize stuff that don't save re-redners. memoize comes in a price of app memory

File: src/incubator/TextField/index.tsx [hunk 2]
```diff
@@ -108,7 +107,7 @@ const TextField = (props: InternalTextFieldProps) => {
             validate={others.validate}
             validationMessage={others.validationMessage}
             validationMessageStyle={validationMessageStyle}
-            retainSpace={retainTopMessageSpace}
```
Comment (ethanshar) [inline src/incubator/TextField/index.tsx:111]: what's the purpose of this change?

**Review by ethanshar — CHANGES_REQUESTED** [review]
@Effanuel 
See my comments, also please fix the PR description. 
In the description section add the PR description
In the changelog add the change for the release notes

File: src/incubator/TextField/index.tsx (hunk 2 above)
Comment (Effanuel) [reply src/incubator/TextField/index.tsx:111]: To avoid calling `isEmpty` and `trim` on every render  if validation message is at the bottom (which is the default). Do you think this is too small of a change worth having?
